cmd/ipfs/daemon.go

```diff
@@ -365,13 +365,26 @@ func daemonFunc(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment
 	defer repo.Close()
 
 	offline, _ := req.Options[offlineKwd].(bool)
-	ipnsps, _ := req.Options[enableIPNSPubSubKwd].(bool)
-	pubsub, _ := req.Options[enablePubSubKwd].(bool)
+	ipnsps, ipnsPsSet := req.Options[enableIPNSPubSubKwd].(bool)
+	pubsub, psSet := req.Options[enablePubSubKwd].(bool)
+
 	if _, hasMplex := req.Options[enableMultiplexKwd]; hasMplex {
 		log.Errorf("The mplex multiplexer has been enabled by default and the experimental %s flag has been removed.")
 		log.Errorf("To disable this multiplexer, please configure `Swarm.Transports.Multiplexers'.")
 	}
 
+	cfg, err := repo.Config()
+	if err != nil {
+		return err
+	}
+
+	if !psSet {
+		pubsub = cfg.Pubsub.Enabled.WithDefault(false)
+	}
+	if !ipnsPsSet {
+		ipnsps = cfg.Ipns.UsePubsub.WithDefault(false)
+	}
```

Comment:
While this will technically do the job of having the daemon flags override the config there's a bit of a weird behavior here in that if we have `--enable-pubsub-experiment=false` and `Ipns.UsePubSub : true` the result will be that pubsub is running.

@lidel any suggestions, does this seem fine as is?

We should probably add tests here that verify things are working as intended

Reply:
@aschmahmann  Added tests at the end of `t0320-pubsub.sh` and `t0183-namesys-pubsub.sh` to confirm that the system stays disabled if cli is `false` and respective config is `true`

Reply:
Documented `--enable-pubsub-experiment=false` in https://github.com/ipfs/go-ipfs/pull/8510/commits/4637863e9ae7ee013b1476dc114994df2c82f473 

I think this is good enough, realistically users will just use config, as they have been asking for it for years.

Reply:
We decided it's fine as is/more confusing to change the behavior